controllers/storagecluster/external_resources.go

```diff
@@ -340,32 +386,41 @@ func (obj *ocsExternalResources) ensureDeleted(r *StorageClusterReconciler, inst
 	return reconcile.Result{}, nil
 }
 
-// claims should be created only once and should not be created/updated again if user deletes/update it.
-func (r *StorageClusterReconciler) createDefaultStorageClassClaims(instance *ocsv1.StorageCluster) error {
-
-	storageClassClaimFile := &ocsv1alpha1.StorageClassClaim{
+// creates claims for rbd and claims should be created only once and should not be created/updated again if user deletes/update it.
+func (r *StorageClusterReconciler) createDefaultStorageClassClaimsForRBD(instance *ocsv1.StorageCluster) error {
+	storageClassClaimBlock := &ocsv1alpha1.StorageClassClaim{
 		ObjectMeta: metav1.ObjectMeta{
-			Name:      generateNameForCephFilesystemSC(instance),
+			Name:      generateNameForCephBlockPoolSC(instance),
 			Namespace: instance.Namespace,
 			Labels: map[string]string{
 				defaultStorageClassClaimLabel: "true",
 			},
 		},
 		Spec: ocsv1alpha1.StorageClassClaimSpec{
-			Type: "sharedfilesystem",
+			Type: "blockpool",
 		},
 	}
 
-	storageClassClaimBlock := &ocsv1alpha1.StorageClassClaim{
+	err := r.createAndOwnStorageClassClaim(instance, storageClassClaimBlock)
+	if err != nil {
+		return err
+	}
+
+	return nil
```

Comment:
just return err?

Reply:
it was like that only, I have just moved it to new method

Reply:
I'll add some log to it

Reply:
I'll add some log to it